.ci.sh

```diff
@@ -48,7 +48,8 @@ check_dependencies() {
 
 run_tests() {
   activate_venv_sct
-  pytest testing/api testing/cli
+  # re-run failed tests, just in case something flaky has happened. if no failures, do nothing.
+  pytest testing/api testing/cli || pytest --last-failed --last-failed-no-failures none
```

Comment:
I've added a command here to re-run failed tests, that also [plays nicely with parametrized tests](https://docs.pytest.org/en/latest/how-to/cache.html#rerunning-only-failures-or-failures-first), too. That way, the `2.0s` threshold will only trigger an error if a script takes longer than 2 seconds in back-to-back runs.